tests/baselines/reference/unionOfClassCalls.types

```diff
@@ -66,7 +66,7 @@ arr.map((a: number | string, index: number) => {
 
 // This case still doesn't work because `reduce` has multiple overloads :(
 arr.reduce((acc: Array<string>, a: number | string, index: number) => { 
->arr.reduce((acc: Array<string>, a: number | string, index: number) => {     return []}, []) : never[]
+>arr.reduce((acc: Array<string>, a: number | string, index: number) => {     return []}, []) : string[]

```

Comment:
Yes, this does fix the `// This case still doesn't work because reduce has multiple overloads :(` comment 😀

Now, since every overload is done in its own speculative context, the `[]` expression no longer has its type locked in by the first overload, and it can be interpreted as desired by the later overload instead.